src/useInView.tsx

```diff
@@ -113,17 +113,23 @@ export function useInView({
   );
 
   const entryTarget = state.entry?.target;
-
-  React.useEffect(() => {
-    if (!ref && entryTarget && !triggerOnce && !skip) {
-      // If we don't have a node ref, then reset the state (unless the hook is set to only `triggerOnce` or `skip`)
-      // This ensures we correctly reflect the current state - If you aren't observing anything, then nothing is inView
-      setState({
-        inView: !!initialInView,
-        entry: undefined,
-      });
-    }
-  }, [ref, entryTarget, triggerOnce, skip, initialInView]);
+  const previousEntryTarget = React.useRef<Element>();
+  if (
+    typeof window !== 'undefined' &&
```

Comment:
Don't think the `window` check is needed, since you wouldn't have the `entryTarget` on the server.